.github/actions/install-rust-toolchain/action.yml

```diff
@@ -0,0 +1,35 @@
+## Inspired by: https://github.com/actions-rs/toolchain/issues/216#issuecomment-1291613319
+name: Install Rust toolchain
+description: "Install Rust toolchain, with caching"
+
+inputs:
+  toolchain:
+    required: false
+    description: "Rust toolchain, e.g. 'stable' or 'nightly'"
+    default: stable
+  cache-key:
+    required: false
+    description: "Extra key to resolve cache"
+    default: ""
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Rust"
+      run: |
+        rustup toolchain install ${{ inputs.toolchain }} --profile minimal --no-self-update
+        rustup default ${{ inputs.toolchain }}
+      shell: bash
+
+    - uses: Swatinem/rust-cache@v2
+      with:
+        shared-key: ${{ inputs.cache-key }}
+
+    - name: "Set environment variables used by toolchain"
+      run: |
+        echo CARGO_TERM_COLOR=always >> $GITHUB_ENV
+        echo CARGO_INCREMENTAL=0 >> $GITHUB_ENV
+        echo RUST_BACKTRACE=1 >> $GITHUB_ENV
```

Comment:
Can we remove this environment variable from the workflow file if specified here?

Reply:
Thanks @TimDiekmann! I addressed your comments in https://github.com/hashintel/hash/pull/1270/commits/2e6e43b08d8c10e777581b6355cbb2b679f5edbd, WDYT?

Reply:
Thanks!